toolkit/tools/imagecustomizer/docs/configuration.md

```diff
@@ -226,6 +235,86 @@ So, any existing ExtraCommandLine value in the base image will be replaced.
 If the partitions are not customized, then the `ExtraCommandLine` value will be appended
 to the existing `grub.cfg` file.
 
+### SELinux
+
+Specifies the mode to set SELinux to.
+
+If this field is not specified, then the existing SELinux mode in the base image is
+maintained.
+Otherwise, the image is modified to match the requested SELinux mode.
+
+The Mariner Image Customizer tool can enable SELinux on a base image with SELinux
+disabled and it can disable SELinux on a base image that has SELinux enabled.
+However, using a base image that already has the required SELinux mode will speed-up the
+customization process.
+
+If SELinux is enabled, then all the file-systems that support SELinux will have their
+file labels updated/reset (using the `setfiles` command).
+
+Supported options:
+
+- `disabled`: Disables SELinux.
+
+- `permissive`: Enables SELinux but only logs access rule violations.
+
+- `enforcing`: Enables SELinux and enforces all the access rules.
+
+- `force-enforcing`: Enables SELinux and sets it to enforcing in the kernel
+  command-line.
+  This means that SELinux can't be set to `permissive` using the `/etc/selinux/config`
+  file.
+
+Note: For images with SELinux enabled, the `selinux-policy` package must be installed.
+This package contains the default SELinux rules and is required for SELinux-enabled
+images to be functional.
+The Mariner Image Customizer tool will report an error if the package is missing from
+the image.
+
+Note: If you wish to apply additional SELinux policies on top of the base SELinux
+policy, then it is recommended to apply these new policies using
+([PostInstallScripts](#postinstallscripts-script)).
+After applying the policues, you do not need to call `setfiles` manually since it will
+called automatically after the `PostInstallScripts` are run.
+
+Example:
+
+```yaml
+SystemConfig:
+  KernelCommandLine:
+    SELinux: enforcing
+
+  PackagesInstall:
+  # Required packages for SELinux.
+  - selinux-policy
+  - selinux-policy-modules
+  
+  # Optional packages that contain useful SELinux utilities.
+  - setools-console
+  - policycoreutils-python-utils
+```
+
+### SELinuxDisablePackageCheck
+
+Disables the check that ensures the `selinux-policy` package is installed when SELinux
+is enabled.
+This is useful for images that want to use their own custom SELinux policy package.
```

Comment:
Question: by allow to disable the check is it possible to misconfigure the image by thinking SELinux is enabled but accidently skipping the check and end up with no policies installed? Do you think it would make sense to replace `SELinuxDisablePackageCheck` with `SELinuxPolicyPackage` and always check for the policy package when SELinux is enabled but when `SELinuxPolicyPackage` is not explicitly set, it defaults to `selinux-policy`?

Related question, since I don't know much about SELinux: is there another way to verify if any SELinux policy exists on the system (in case it's installed through adding single files or post-install scripts but not through any RPM package)?

Reply:
The `selinux-policy` package provides the `/etc/selinux/config` file. This file specifies what policy to use and is required for SELinux to work. So, I can check for existence of that file. This is probably a better solution than looking for an installed package.